server/store/kv_store.go

```
@@ -0,0 +1,132 @@
+package store
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	storeTokenSuffix          = "_circleci_token"  // Full key format is userID_circleci_token
+	defaultProjectStoreSuffix = "_default_project" // Full key format is userID_default_project
+	subscriptionsKVKey        = "subscriptions"
+)
+
+// GetTokenForUser returns false if no token is saved for this user
+func (s *Store) GetTokenForUser(userID, encryptionKey string) (string, bool) {
+	raw, appErr := s.api.KVGet(userID + storeTokenSuffix)
+	if appErr != nil {
+		s.api.LogError("Unable to reach KVStore", "KVStore error", appErr)
+		return "", false
+	}
+
+	if raw == nil {
+		return "", false
+	}
+
+	userToken, err := decrypt([]byte(encryptionKey), string(raw))
+	if err != nil {
+		s.api.LogWarn("Failed to decrypt access token", "error", err)
+		return "", false
+	}
+
+	return userToken, true
+}
+
+// StoreTokenForUser returns false if the token has not been saved
+func (s *Store) StoreTokenForUser(userID, circleciToken, encryptionKey string) bool {
```

In general, I would return an error vs a bool. I see some other methods which do this, such as `GetSubscriptions`

---

I tried to hunt all of them and changed them to errors :white_check_mark: 
I'm not sure if the changes make the code more readable, you can check it [here`8d266862`](https://github.com/mattermost/mattermost-plugin-circleci/pull/69/commits/8d266862966deaf609afcd74b66320d332031a04).

---

Looks great!